### PR TITLE
Add cloud apps folder and initial query

### PR DESCRIPTION
**Pull request by Keithbfleming**
Required items, please complete
   
   Change(s):
   - Added folder for Cloud Apps in M365 Hunting and an initial query for file download events

   Reason for Change(s):
   - No existing sample queries for MDA

   Version Updated:
   - N/A

   Testing Completed:
   - Validated in M365D AH

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by v-rbajaj** [comment]
Thank you for raising Pull Request with us! You should expect an initial response to your Pull Request from the team within 2 business days. Note that this response may be delayed during holiday periods.

File: Hunting Queries/Microsoft 365 Defender/Cloud Apps/file-download-events.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+id: f8f8d4a5-7d7d-4c5d-9b5c-9c5d7d8f8f8f
+
+name: File download events in the last 7 days
+
+description: |
+  This query looks for file download events identified by Microsoft Defender for Cloud Apps. It will require an corresponding app connector in Microsoft Defender for Cloud Apps.
+  Reference - https://learn.microsoft.com/en-us/defender-cloud-apps/enable-instant-visibility-protection-and-governance-actions-for-your-apps
+  Reference - https://github.com/Azure/Azure-Sentinel/blob/master/Hunting%20Queries/Cloud Apps/file-download-events.yaml
```
Comment (v-rbajaj) [inline Hunting Queries/Microsoft 365 Defender/Cloud Apps/file-download-events.yaml:8]: Hi @Keithbfleming, second reference link isn't working.

File: Hunting Queries/Microsoft 365 Defender/Cloud Apps/file-download-events.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+id: f8f8d4a5-7d7d-4c5d-9b5c-9c5d7d8f8f8f
+
+name: File download events in the last 7 days
+
+description: |
+  This query looks for file download events identified by Microsoft Defender for Cloud Apps. It will require an corresponding app connector in Microsoft Defender for Cloud Apps.
+  Reference - https://learn.microsoft.com/en-us/defender-cloud-apps/enable-instant-visibility-protection-and-governance-actions-for-your-apps
```
Comment (v-rbajaj) [inline Hunting Queries/Microsoft 365 Defender/Cloud Apps/file-download-events.yaml:7]: Can you please do not include language specific links?

**Comment by v-rbajaj** [comment]
Please do add the `version`

Refer this HQ :- https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Azure%20Activity/Hunting%20Queries/AnalyticsRulesAdministrativeOperations.yaml

File: Hunting Queries/Microsoft 365 Defender/Cloud Apps/file-download-events.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  Reference - https://github.com/Azure/Azure-Sentinel/blob/master/Hunting%20Queries/Cloud Apps/file-download-events.yaml
+requiredDataConnectors:
+- connectorId: MicrosoftThreatProtection
+  dataTypes:
+  - CloudAppEvents
+tactics:
+- Exfiltration
+query: >
+  let LookBack = 7d;
```
[the remark this answers is not in the document]
Comment (Keithbfleming) [reply Hunting Queries/Microsoft 365 Defender/Cloud Apps/file-download-events.yaml:16]: @v-rbajaj  without the > the YAML didn't validate in a checker which is why it's here

File: Hunting Queries/Microsoft 365 Defender/Cloud Apps/file-download-events.yaml (hunk 1 above)
Comment (Keithbfleming) [reply Hunting Queries/Microsoft 365 Defender/Cloud Apps/file-download-events.yaml]: @v-rbajaj it's a link to the yaml itself, I wouldn't expect it to work until committed in the master branch

File: Hunting Queries/Microsoft 365 Defender/Cloud Apps/file-download-events.yaml (hunk 2 above)
Comment (Keithbfleming) [reply Hunting Queries/Microsoft 365 Defender/Cloud Apps/file-download-events.yaml]: @v-rbajaj  removed in last commit

**Review by Keithbfleming — COMMENTED** [review]
Added entity mappings and version

File: Hunting Queries/Microsoft 365 Defender/Cloud Apps/file-download-events.yaml (hunk 1 above)
Comment (v-rbajaj) [reply Hunting Queries/Microsoft 365 Defender/Cloud Apps/file-download-events.yaml]: I think you are trying to reference this link - https://github.com/Azure/Azure-Sentinel/blob/master/Hunting%20Queries/Microsoft%20365%20Defender/Cloud%20Apps/file-download-events.yaml

**Comment by Keithbfleming** [comment]
That’s correct.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by v-rbajaj** [comment]
> Added entity mappings and version

Hi @Keithbfleming, Thanks for making these however the indentation for entitymappings isn't correct,

Apart from it, in the query you need to extend the columns this way.

```
| extend Account_0_ObjectGuid = AccountObjectId
| extend Account_0_DisplayName = AccountDisplayName
```

**Comment by Keithbfleming** [comment]
What’s the proper indentation syntax for entityMapping and is it required here?

To clarify calculated columns that are extended need to be on separate lines?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by v-rbajaj** [comment]
Hi @Keithbfleming, So please correct the second reference link as mentioned above. 

For correct indentation of entitymappings please refer to Hunting Query :- https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Azure%20Activity/Hunting%20Queries/AnalyticsRulesAdministrativeOperations.yaml

Yes, columns that are extended need to be on separate lines.

**Comment by Keithbfleming** [comment]
Hi,

The link has now been completed removed, I'll add a new PR for this later when the file exists and can be referenced.

EntityMapping was also updated.
________________________________

**Comment by v-atulyadav** [comment]
Hi @Keithbfleming,
Thank you for accommodating the suggested changes! We will review the Pull Request and get back to you by 16 Aug 2023.

**Review by v-rbajaj — APPROVED** [review]

**Pull request merged**